service/controller/v2/resource/chart/resource.go

```diff
@@ -69,6 +71,21 @@ func (r *Resource) Name() string {
 	return Name
 }
 
+func isChartModified(a, b ChartState) bool {
+	// ChartValues have changed so we need to update.
```

Comment:
Nitpick, we need to update if the next `if` condition evaluates to true, is that correct?

Reply:
Good point. I'm going to make the ReleaseVersion check first. As if it has changed we need to update the Helm release with the new chart version. 

If the values have changed we just need to update them for the current Helm release. I'll update the comments to make them clearer.

Reply:
Done.